File: decompiled/sources/com/hjq/toast/ToastUtils.java
package com.hjq.toast;

import android.app.Application;
import android.content.res.Resources;
import com.hjq.toast.config.IToastInterceptor;
import com.hjq.toast.config.IToastStrategy;
import com.hjq.toast.config.IToastStyle;
import com.hjq.toast.style.BlackToastStyle;
import com.hjq.toast.style.LocationToastStyle;
import com.hjq.toast.style.ViewToastStyle;

public final class ToastUtils {
    private static Application sApplication;
    private static Boolean sDebugMode;
    private static IToastInterceptor sToastInterceptor;
    private static IToastStrategy sToastStrategy;
    private static IToastStyle<?> sToastStyle;

    private ToastUtils() {
    }

    public static void init(Application application) {
        init(application, sToastStyle);
    }

    public static void init(Application application, IToastStyle<?> iToastStyle) {
        sApplication = application;
        if (sToastStrategy == null) {
            setStrategy(new ToastStrategy());
        }
        if (iToastStyle == null) {
            iToastStyle = new BlackToastStyle();
        }
        setStyle(iToastStyle);
    }

    public static boolean isInit() {
        return (sApplication == null || sToastStrategy == null || sToastStyle == null) ? false : true;
    }

    public static void show(Object obj) {
        show((CharSequence) obj != null ? obj.toString() : "null");
    }

    public static void debugShow(Object obj) {
        if (isDebugMode()) {
            show(obj);
        }
    }

    public static void show(int i) {
        try {
            show(sApplication.getResources().getText(i));
        } catch (Resources.NotFoundException unused) {
            show((CharSequence) String.valueOf(i));
        }
    }

    public static void debugShow(int i) {
        if (isDebugMode()) {
            show(i);
        }
    }

    public static void show(CharSequence charSequence) {
        if (charSequence != null && charSequence.length() != 0) {
            IToastInterceptor iToastInterceptor = sToastInterceptor;
            if (iToastInterceptor == null || !iToastInterceptor.intercept(charSequence)) {
                sToastStrategy.showToast(charSequence);
            }
        }
    }

    public static void debugShow(CharSequence charSequence) {
        if (isDebugMode()) {
            show(charSequence);
        }
    }

    public static void cancel() {
        sToastStrategy.cancelToast();
    }

    public static void setGravity(int i) {
        setGravity(i, 0, 0);
    }

    public static void setGravity(int i, int i2, int i3) {
        setGravity(i, i2, i3, 0.0f, 0.0f);
    }

    public static void setGravity(int i, int i2, int i3, float f, float f2) {
        sToastStrategy.bindStyle(new LocationToastStyle(sToastStyle, i, i2, i3, f, f2));
    }

    public static void setView(int i) {
        if (i > 0) {
            setStyle(new ViewToastStyle(i, sToastStyle));
        }
    }

    public static void setStyle(IToastStyle<?> iToastStyle) {
        sToastStyle = iToastStyle;
        sToastStrategy.bindStyle(iToastStyle);
    }

    public static IToastStyle<?> getStyle() {
        return sToastStyle;
    }

    public static void setStrategy(IToastStrategy iToastStrategy) {
        sToastStrategy = iToastStrategy;
        iToastStrategy.registerStrategy(sApplication);
    }

    public static IToastStrategy getStrategy() {
        return sToastStrategy;
    }

    public static void setInterceptor(IToastInterceptor iToastInterceptor) {
        sToastInterceptor = iToastInterceptor;
    }

    public static IToastInterceptor getInterceptor() {
        return sToastInterceptor;
    }

    public static void setDebugMode(boolean z) {
        sDebugMode = Boolean.valueOf(z);
    }

    private static boolean isDebugMode() {
        if (sDebugMode == null) {
            sDebugMode = Boolean.valueOf((sApplication.getApplicationInfo().flags & 2) != 0);
        }
        return sDebugMode.booleanValue();
    }
}

File: decompiled/sources/com/hjq/toast/ToastUtils.kt
package com.hjq.toast

import android.app.Application
import android.content.res.Resources
import com.hjq.toast.config.IToastInterceptor
import com.hjq.toast.config.IToastStrategy
import com.hjq.toast.config.IToastStyle
import com.hjq.toast.style.BlackToastStyle
import com.hjq.toast.style.LocationToastStyle
import com.hjq.toast.style.ViewToastStyle

class ToastUtils private constructor() {
    companion object {
        private var sApplication: Application? = null
        private var sDebugMode: Boolean? = null
        private var sToastInterceptor: IToastInterceptor? = null
        private var sToastStrategy: IToastStrategy? = null
        private var sToastStyle: IToastStyle<*>? = null

        @JvmStatic
        fun init(application: Application) {
            init(application, sToastStyle)
        }

        @JvmStatic
        fun init(application: Application, style: IToastStyle<*>?) {
            sApplication = application
            if (sToastStrategy == null) {
                setStrategy(ToastStrategy())
            }
            val toastStyle = style ?: BlackToastStyle()
            setStyle(toastStyle)
        }

        @JvmStatic
        fun isInit(): Boolean = sApplication != null && sToastStrategy != null && sToastStyle != null

        @JvmStatic
        fun show(obj: Any?) {
            show(obj?.toString() ?: "null")
        }

        @JvmStatic
        fun debugShow(obj: Any?) {
            if (isDebugMode()) {
                show(obj)
            }
        }

        @JvmStatic
        fun show(textResId: Int) {
            try {
                show(sApplication!!.resources.getText(textResId))
            } catch (_: Resources.NotFoundException) {
                show(textResId.toString())
            }
        }

        @JvmStatic
        fun debugShow(textResId: Int) {
            if (isDebugMode()) {
                show(textResId)
            }
        }

        @JvmStatic
        fun show(text: CharSequence?) {
            if (!text.isNullOrEmpty()) {
                val toastInterceptor = sToastInterceptor
                if (toastInterceptor == null || !toastInterceptor.intercept(text)) {
                    sToastStrategy!!.showToast(text)
                }
            }
        }

        @JvmStatic
        fun debugShow(text: CharSequence?) {
            if (isDebugMode()) {
                show(text)
            }
        }

        @JvmStatic
        fun cancel() {
            sToastStrategy!!.cancelToast()
        }

        @JvmStatic
        fun setGravity(gravity: Int) {
            setGravity(gravity, 0, 0)
        }

        @JvmStatic
        fun setGravity(gravity: Int, xOffset: Int, yOffset: Int) {
            setGravity(gravity, xOffset, yOffset, 0.0f, 0.0f)
        }

        @JvmStatic
        fun setGravity(gravity: Int, xOffset: Int, yOffset: Int, horizontalMargin: Float, verticalMargin: Float) {
            sToastStrategy!!.bindStyle(
                LocationToastStyle(sToastStyle, gravity, xOffset, yOffset, horizontalMargin, verticalMargin)
            )
        }

        @JvmStatic
        fun setView(layoutId: Int) {
            if (layoutId > 0) {
                setStyle(ViewToastStyle(layoutId, sToastStyle))
            }
        }

        @JvmStatic
        fun setStyle(style: IToastStyle<*>) {
            sToastStyle = style
            sToastStrategy!!.bindStyle(style)
        }

        @JvmStatic
        fun getStyle(): IToastStyle<*>? = sToastStyle

        @JvmStatic
        fun setStrategy(strategy: IToastStrategy) {
            sToastStrategy = strategy
            strategy.registerStrategy(sApplication!!)
        }

        @JvmStatic
        fun getStrategy(): IToastStrategy? = sToastStrategy

        @JvmStatic
        fun setInterceptor(interceptor: IToastInterceptor?) {
            sToastInterceptor = interceptor
        }

        @JvmStatic
        fun getInterceptor(): IToastInterceptor? = sToastInterceptor

        @JvmStatic
        fun setDebugMode(debugMode: Boolean) {
            sDebugMode = debugMode
        }

        private fun isDebugMode(): Boolean {
            if (sDebugMode == null) {
                sDebugMode = (sApplication!!.applicationInfo.flags and 2) != 0
            }
            return sDebugMode == true
        }
    }
}
